src/main/scala/org/scalameter/persistence/codecs/domain.scala

```diff
@@ -0,0 +1,111 @@
+        measurements = jObj.get("measurements").asInstanceOf[JArray].vs.map(MeasurementCodec.unsafeDecode),
+        info = KeyMapCodec.unsafeDecode(jObj.get("info")),
+        context = ContextCodec.unsafeDecode(jObj.get("context"))
+      )
+      case _ => sys.error(s"Expected JObject in CurveDataCodec. Got: $jV.")
+    }
+  }
+}
+
+object MeasurementCodec extends JSONCodec[Measurement] {
+  def encode(obj: Measurement): JValue = JObject.fromSeq(Seq(
+    "value" -> DoubleCodec.encode(obj.value),
+    "params" -> ParametersCodec.encode(obj.params),
+    "data" -> MeasurementDataCodec.encode(obj.data),
+    "units" -> StringCodec.encode(obj.units)
+  ))
+
+  def decode(jV: JValue)(implicit hint: TypeHint): Try[Measurement] = Try {
+    jV match {
+      case jObj: JObject => Measurement(
+        DoubleCodec.unsafeDecode(jObj.get("value")),
+        ParametersCodec.unsafeDecode(jObj.get("params")),
+        MeasurementDataCodec.unsafeDecode(jObj.get("data")),
+        StringCodec.unsafeDecode(jObj.get("units"))
+      )
+      case _ => sys.error(s"Expected JObject in ParametersCodec. Got: $jV.")
+    }
+  }
+}
+
+object ParametersCodec extends JSONCodec[Parameters] {
+  def encode(obj: Parameters): JValue = JObject.fromSeq(Seq(
+    "axisData" -> ListMapCodec.encode(obj.axisData)
+  ))
+
+  def decode(jV: JValue)(implicit hint: TypeHint): Try[Parameters] = Try {
+    jV match {
+      case jObj: JObject => Parameters(ListMapCodec.unsafeDecode(jObj.get("axisData")))
+      case _ => sys.error(s"Expected JObject in ParametersCodec. Got: $jV.")
+    }
+  }
+}
+
+object MeasurementDataCodec extends JSONCodec[MeasurementData] {
+  def encode(obj: MeasurementData): JValue = JObject.fromSeq(Seq(
+    "complete" -> JArray.fromSeq(obj.complete.map(DoubleCodec.encode)),
+    "success" -> BooleanCodec.encode(obj.success)
+  ))
+
+  def decode(jV: JValue)(implicit hint: TypeHint): Try[MeasurementData] = Try {
+    jV match {
+      case jObj: JObject => MeasurementData(
+        jObj.get("complete").asInstanceOf[JArray].vs.map(DoubleCodec.unsafeDecode)(collection.breakOut),
+        BooleanCodec.unsafeDecode(jObj.get("success"))
+      )
+      case _ => sys.error(s"Expected JObject in ParametersCodec. Got: $jV.")
+    }
+  }
+}
+
+object ResultCodec extends Tuple3Codec[Date, Context, CurveData]
+
+object HistoryCodec extends JSONCodec[History] {
+  def encode(history: History): JValue = {
+    JObject.fromSeq(List(
+      "results" -> JArray.fromSeq(history.results.map(ResultCodec.encode)),
+      "infomap" -> KeyMapCodec.encode(history.infomap)
+    ))
+  }
+
+  def decode(jV: JValue)(implicit hint: TypeHint): Try[History] = Try {
+    jV match {
+      case jObj: JObject => History(
+        results = jObj.get("results").asInstanceOf[JArray].vs.map(ResultCodec.unsafeDecode),
+        infomap = KeyMapCodec.unsafeDecode(jObj.get("infomap"))
+      )
+      case _ => sys.error(s"Expected JObject in HistoryCodec. Got: $jV.")
+    }
+  }
+}
```

Comment:
It seems like: if you add a new field to some class,
you might forget to update the `decode` method for it. Fortunately, most of our classes have all the fields in the constructor, so that would be detected during compilation, so I don't think it is a huge problem.

One bigger potential issue is that the codecs write potentially the same objects both when they write to JSON, and recreate them when they write back. If the `Context`, which could take a lot of space, is duplicated for each curve (and curves typically share the same `Context`  object), it takes up a lot of space. Perhaps it would be useful to cache some of these objects during a codec run.